File: apps/client/src/common/types/invoices.ts
import { SqlTable } from './sql';

export type ItemInvoiceT = SqlTable<{
  name: string;
  qty: number;
  price: number;
  invoiceId: number;
}>;

export type PaymentTermT = 'Net 1 Day' | 'Net 7 Days' | 'Net 14 Days' | 'Net 30 Days';

export type InvoiceStatusT = 'DRAFT' | 'PENDING' | 'PAID';

export type InvoiceT = SqlTable<{
  clientId: string;

  billFromStreet: string;
  billFromCity: string;
  billFromZip: string;
  billFromCountry: string;

  billToClientName: string;
  billToClientMail: string;
  billToStreet: string;
  billToCity: string;
  billToZip: string;
  billToCountry: string;

  invoiceDate: string;
  paymentTerm: PaymentTermT;
  description: string;
  status: InvoiceStatusT;

  itemsList: ItemInvoiceT[];
}>;

File: apps/server/Models/Invoices.cs
public class Invoices
{
  public int Id { get; set; }

  public string BillFromStreet { get; set; } = null!;
  public string BillFromCity { get; set; } = null!;
  public string BillFromZip { get; set; } = null!;
  public string BillFromCountry { get; set; } = null!;
  public string BillToClientName { get; set; } = null!;
  public string BillToClientMail { get; set; } = null!;
  public string BillToStreet { get; set; } = null!;
  public string BillToCity { get; set; } = null!;
  public string BillToZip { get; set; } = null!;
  public string BillToCountry { get; set; } = null!;
  public DateTime InvoiceDate { get; set; }
  public string PaymentTerm { get; set; } = null!;
  public string Description { get; set; } = null!;
  public string Status { get; set; } = null!;

  public List<ItemsList> Items { get; set; } = [];
}